packages/flutter_tools/test/src/mocks.dart

```diff
@@ -108,6 +108,48 @@ ro.build.version.codename=REL
 ''';
 }
 
+/// A broken SDK installation.
+class MockBrokenAndroidSdk extends Mock implements AndroidSdk {
```

Comment:
Just put this in `android_sdk_test.dart` since it's not likely to be used outside that file.